contracts/MesaFactory.sol

```diff
@@ -31,43 +29,48 @@ contract MesaFactory {
     address public templateManager;
     address public templateLauncher;
     uint256 public templateFee;
-    address[] public allSales;
+
+    address[] public allTemplates;
     uint256 public templateId;
-    bool initialized = false;
+    bool public initialized = false;
 
-    constructor() public {}
+    modifier isTemplateManager {
+        require(msg.sender == templateManager, "MesaFactory: FORBIDDEN");
+        _;
+    }
+
+    modifier isFeeManager {
+        require(msg.sender == feeManager, "MesaFactory: FORBIDDEN");
+        _;
+    }
 
     /// @dev setup function to initialize the Mesa Factory
     /// @param _feeManager address that is allowed to update fees
     /// @param _feeTo address that receives fees
     /// @param _templateManager address that is allowed to manage templates
-    /// @param _templateLauncher address of the template launcher used to launch projects
     /// @param _templateFee fixed amount of native currency (ETH) to be paid for adding a template
     /// @param _feeNumerator fee that is token on depositing tokens
     /// @param _saleFee fixed amount of native currency (ETH) to be paid for launch a project
-    function initialize(
+    constructor(
```

Comment:
Any specific reason why we opt into the `constructor` to initialize the factory?

Reply:
With the initialize function additional logic is needed to handle the state as the initialized = true/false to prevent reentrancies & make sure only the owner is able to initialize. The constructor handles this natively, so we can simplify here a bit.